### PR TITLE
Document (new) JS APIs

### PR DESCRIPTION
**Pull request by ChristophWurst**
Ref https://github.com/nextcloud/server/issues/15932
Ref https://github.com/nextcloud/server/issues/18677

### CODE REVIEW

File: developer_manual/app/javascript-apis.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+This package provides helpers to generate URLs, e.g. to access assets and REST APIs of your app or the Nextcloud server.
+`Read the full documentation <https://nextcloud.github.io/nextcloud-router/>`_
+
+
+``@nextcloud/vue``
+^^^^^^^^^^^^^^^^^^
+
+This package provides a component library.
+`Read the full documentation <https://github.com/nextcloud/nextcloud-vue/#documentation>`_
```
Comment (juliushaertl) [inline developer_manual/app/javascript-apis.rst]: I'd say we can link directly to https://nextcloud-vue-components.netlify.com/

Comment (ChristophWurst) [reply developer_manual/app/javascript-apis.rst]: sure! I wasn't sure if this was the final domain. but we of course always update the docs if the domain changes :)

**Comment by skjnldsv** [comment]
I disagree with the OCA part. Every app that register is fine. But any app that uses them needs to wait for domcontentloaded, this works fine and it is really the only way to expose proper APIs like the viewer or the sidebar without too much annoyance I think :)

**Comment by ChristophWurst** [comment]
> I disagree with the OCA part. Every app that register is fine. But any app that uses them needs to wait for domcontentloaded, this works fine and it is really the only way to expose proper APIs like the viewer or the sidebar without too much annoyance I think :)

I know, thus

> Except for edge cases with inter-app communication, this use of this approach is discouraged as well

Apps used to assign all their Backbone models and views to OCA. That should go away. For the edge case like with viewer it's fine, though we could hide the global variable in a tiny wrapper package.

**Comment by skjnldsv** [comment]
> > Except for edge cases with inter-app communication, this use of this approach is discouraged as well
> 
> Apps used to assign all their Backbone models and views to OCA. That should go away. For the edge case like with viewer it's fine, though we could hide the global variable in a tiny wrapper package.

Yeah, I thought is was not very clear. We should maybe be more specifics?

**Comment by ChristophWurst** [comment]
> Yeah, I thought is was not very clear. We should maybe be more specifics?

Please see updated text.

File: developer_manual/app/javascript-apis.rst [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+^^^^^^^^^^^^^^^^^^^^^
+
+This package provides helpers to generate URLs, e.g. to access assets and REST APIs of your app or the Nextcloud server.
+`Read the full documentation <https://nextcloud.github.io/nextcloud-router/>`_
+
+
+``@nextcloud/vue``
+^^^^^^^^^^^^^^^^^^
+
+This package provides a component library.
```
Comment (skjnldsv) [inline developer_manual/app/javascript-apis.rst]:
```suggestion
This package provides lots of nextcloud components allowing you to quickly build UIs.
```
? Or something?

Comment (ChristophWurst) [reply developer_manual/app/javascript-apis.rst]: sure :)

**Comment by ChristophWurst** [comment]
:pray:

**Pull request merged**